include/nbl/builtin/hlsl/random/lcg.hlsl

```diff
@@ -0,0 +1,34 @@
+// Copyright (C) 2018-2025 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_GLSL_RANDOM_LCG_HLSL_INCLUDED_
+#define _NBL_BUILTIN_GLSL_RANDOM_LCG_HLSL_INCLUDED_
+
+namespace nbl
+{
+namespace hlsl
+{
+
+struct Lcg
+{
+    static Lcg construct(NBL_CONST_REF_ARG(uint32_t) state)
+	{
+        return Lcg(state);
```

Comment:
ALWAYS use the 
```
T retval;
retval.someMember = ...;
return retval;
```
pattern.

don't leave yourself at the mercy of HLSL's broken conversion operators and initialization.

Reply:
done by me